### DISCUSSION TITLE
RP2040 + ST7796 + XPT2046 - Touch values are very odd and I can't get no reason for that

### DISCUSSION THREAD

**Discussion by gymho-gabriel** [Bodmer/TFT_eSPI#3147] [Q&A - General] [answered]
Hey community,
after spending 4 hours with this specific issue (maybe only my issue, not TFT_eSPI's) I am asking you. A similar 

I am using a cheaper 4'' display (with SD-reader, but not connected) - [LCDWiki-Displaymodule](http://www.lcdwiki.com/4.0inch_SPI_Module_ST7796)
My circuit only consist of a raspi pico and the display.

Connections:
`TFT       PICO

VCC     - 3V3
GND     - GND
CS      - GP13
RESET   - 3V3
DataCmd - GP17
MOSI    - SPI1 TX  GP11
SCK     - SPI1 CLK GP10
LED     - GP16
MISO    - N/C

T_CLK   - SPI1 CLK GP10
T_CS    - GP15
T_DIN   - SPI1 TX GP11
T_DO    - SPI1 RX GP12
T_IRQ   - N/C`

i did not connect MISO on display side and the IRQ pin on the touch side.

My Pin Setup in user_setup.h is as follow:
`#define USER_SETUP_INFO "RP2040-ST7796"
#define ST7796_DRIVER
#define TFT_WIDTH 320
#define TFT_HEIGHT 480 
// #define TFT_INVERSION_ON
// #define TFT_INVERSION_OFF
#define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)
#define TFT_MISO  12
#define TFT_MOSI  11
#define TFT_SCLK  10
#define TFT_CS    13  // Chip select control pin D8
#define TFT_DC    14  // Data Command control pin
#define TFT_RST   -1  // Set TFT_RST to -1 if the display RESET is connected to NodeMCU RST or 3.3V
#define TFT_BL    16  // LED back-light (only for ST7789 with backlight control pin)
#define TOUCH_CS  15  // Chip select pin (T_CS) of touch screen

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT

#define TFT_SPI_PORT 1
#define SPI_FREQUENCY  10000000
#define SPI_TOUCH_FREQUENCY  1000000
//#define SUPPORT_TRANSACTIONS`

i experimented a lot with SPI_FREQUENCIES...

**So what is the deal with my setup?**

when leaving the display untouched, the values show (with Test_Touch_Controller):
`x: 0  y:7936  raw:62463
x: 0  y:7936  raw:62463
x: 0  y:7936  raw:65535
x: 0  y:7936  raw:4351
x: 0  y:7936  raw:62207
x: 0  y:7936  raw:62207
x: 0  y:7936  raw:4607
x: 0  y:7936  raw:4863
x: 0  y:7936  raw:62463
x: 0  y:7936  raw:64255
x: 0  y:7936  raw:61695
x: 0  y:7936  raw:62975
x: 0  y:7936  raw:5119
x: 0  y:7936  raw:62719
x: 0  y:7936  raw:62207
x: 0  y:7936  raw:3071
x: 0  y:7936  raw:5119
x: 0  y:7936  raw:61695
x: 0  y:7936  raw:64255
x: 0  y:7936  raw:4863`

When I touch the area: 
`x: 768  y:7936  raw:8959
x: 4096  y:4096  raw:4607
x: 3840  y:2048  raw:1791
x: 2048  y:0  raw:0
x: 7936  y:2048  raw:12031
x: 2048  y:0  raw:10239
x: 3840  y:7936  raw:9983
x: 7936  y:4096  raw:2815
x: 0  y:2048  raw:65535
x: 0  y:4096  raw:8703
x: 7936  y:0  raw:63743
x: 4096  y:4096  raw:3839
x: 2048  y:2048  raw:3839
x: 6912  y:0  raw:4607
x: 0  y:2304  raw:255
x: 3840  y:7168  raw:6143
x: 7936  y:4096  raw:5887`

so this is one fingerpress - several seconds. 

Ideas, guys? I do not know what it is. Anyway, Bodmer, the library is great! I think i am too stupid.

Regards,
Peter

EDIT: changed hardware and settings to SPI Bus Number 0. No Change!

EDIT: Changed BSP from Arduino MBED Pico to Earle Philhower's. This changed a lot. Y Values are still > 2000 when untouched. The biggest change ist touch RAW - this value stays below 200.
Experimented with differen Frequency-Values - does not have much effekt. The goal ist to keep both x and y to zero when left untouched. Then the calibration routine will not automatically go through the process...

**Comment by gymho-gabriel** [comment, marked as answer]
Alright - so anything is odd - try to use Earle's BSP instead of the shipped one in Arduino IDE

**Comment by OlivierCROISET** [comment]
Connection below that works :
``
// SPI port 1 to the TFT screen
#define TFT_MISO   12  //  pin 16
#define TFT_SCK     14  //  pin 19
#define TFT_MOSI   15  //  pin 20
#define TFT_DC        7  //  pin 10 
#define TFT_RST     99  //   unused pin
#define TFT_CS       13  //  pin 17
Adafruit_ILI9341 tft( TFT_CS, TFT_DC, TFT_MOSI, TFT_SCK, TFT_RST, TFT_MISO);

// Connect the touchpad to the **default** SPI0 port
#include <XPT2046.h>
// Default SPI port to the touchpad
//#define TOUCH_SCK     10    // pin 14
#define TOUCH_CS_PIN   17    // pin 22
//#define TOUCH_MOSI   19    // pin 25
//#define TOUCH_MISO   16    // pin 21
#define TOUCH_IRQ_PIN  98    //  unused pin

XPT2046 touch(TOUCH_CS_PIN, TOUCH_IRQ_PIN);
``
Both TFT screen and touchpad will work correctly !
